### PR TITLE
Adding method getSelectionBounds + unittest

### LINKED ISSUE #1266: getSelectionBounds for StaticGlyphController
I think we're still only using the on-curve points here.

Ideally, I think StaticGlyphController should get a method `getSelectionBounds(selection)` that would do the right thing given our "sparse" selection.

Maybe we should do this as a follow-up PR, and not let this improvement delay the landing of this PR.

_Originally posted by @justvanrossum in https://github.com/googlefonts/fontra/pull/1247#discussion_r1558998755_

### PR DESCRIPTION
**Pull request by ollimeier**
Fixes #1266

### CODE REVIEW

File: src/fontra/client/core/glyph-controller.js [hunk 1]
```diff
@@ -619,6 +620,35 @@ export class StaticGlyphController {
...
+
+    for (const componentIndex of componentIndices) {
+      const component = this.components[componentIndex];
+      if (!component || !component.bounds) {
+        continue;
+      }
+      selectionRects.push(component.bounds);
+    }
+
+    if (selectionRects.length) {
```
Comment (justvanrossum) [inline src/fontra/client/core/glyph-controller.js:647]: You can unconditionally do
```js
    return unionRect(...selectionRects);
```
As unionRect can handle an empty list.

File: src/fontra/views/editor/panel-transformation.js [hunk 2]
```diff
@@ -316,38 +316,9 @@ export default class TransformationPanel extends Panel {
...
-    }
-  }
-
-  _getPinPoint(layerGlyphController, pointIndices, componentIndices, originX, originY) {
-    let bounds = this._getSelectedBounds(
-      layerGlyphController,
-      pointIndices,
-      componentIndices
+  _getPinPoint(layerGlyphController, originX, originY) {
+    let bounds = layerGlyphController.getSelectionBounds(
```
Comment (justvanrossum) [inline src/fontra/views/editor/panel-transformation.js:320]:
```suggestion
    const bounds = layerGlyphController.getSelectionBounds(
```

File: src/fontra/client/core/glyph-controller.js (hunk 1 above)
Comment (ollimeier) [reply src/fontra/client/core/glyph-controller.js:647]: done.

Comment (justvanrossum) [reply src/fontra/client/core/glyph-controller.js:647]: With "unconditionally" I meant you can leave out the "if" statement. There's no need for the .length check.

Comment (ollimeier) [reply src/fontra/client/core/glyph-controller.js]: Sorry. Changed it.

**Review by justvanrossum — APPROVED** [review]
Thank you!

**Pull request merged**